lib/blocking_queue.ex

```diff
@@ -71,6 +71,7 @@ defmodule BlockingQueue do
   @spec handle_call(call_t, from_t, state_t) :: result_t
 
   def handle_call({:push, item}, waiter, {max, queue={left,right}}) when length(left) + length(right) >= max do
+    GenServer.reply(waiter, :block)
     {:noreply, {max, queue, :push, waiter, item}}
```

Comment:
You shouldn't need both `GenServer.reply` and `{:noreply, ...}`.  Probably best would be to fill in the return value with  `{:reply, :block, {max, queue, :push, waiter, item}}`.  See 

  http://elixir-lang.org/docs/stable/elixir/GenServer.html#c:handle_call/3

for more details.